platforms/ide/problems-api/src/main/java/org/gradle/api/problems/UnboundBasicProblemBuilder.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.problems;
+
+import org.gradle.api.Incubating;
+import org.gradle.util.Path;
+
+import javax.annotation.Nullable;
+
+/**
+ * Problem builder creating basic, non-reportable problems without enforcing any particular invocation order.
+ *
+ * @since 8.6
+ */
+@Incubating
+public interface UnboundBasicProblemBuilder extends UnboundProblemBuilder, BasicProblemBuilder {
+    UnboundBasicProblemBuilder documentedAt(DocLink doc);
+    UnboundBasicProblemBuilder undocumented();
+    UnboundBasicProblemBuilder fileLocation(String path, @Nullable Integer line, @Nullable Integer column, @Nullable Integer length);
+    UnboundBasicProblemBuilder pluginLocation(String pluginId);
+    UnboundBasicProblemBuilder stackLocation();
+    UnboundBasicProblemBuilder noLocation();
+    UnboundBasicProblemBuilder taskPathLocation(Path location);
```

Comment:
`taskPathLocation` is not available on the the stepwise builder, so we are exposing something that's not even documented. I'd rather have some consistency. We have a few options:
- Adding taskPathLocation to the stepwise builder
- Removing taskPathLocation from the public api and fix the callsite with a cast to DefaultProblemBuilder.

Reply:
I removed the taskPathLocation from the public API